src/irc/core/irc-servers.c

```diff
@@ -586,33 +588,60 @@ char *irc_server_get_channels(IRC_SERVER_REC *server)
 	GSList *tmp;
 	GString *chans, *keys;
 	char *ret;
+	rejoin_channels_mode = g_strdup(settings_get_str("rejoin_channels_on_reconnect"));
```

Comment:
Leak here the second time this function is called. Blame the scope of the variable, which as far as i can see isn't needed. Also I don't think the g_strdup is needed at all.